extensions/wikia/UserRenameTool/UserRenameToolProcess.class.php

```diff
@@ -246,197 +194,358 @@ protected function renameAccount() {
 		return false;
 	}
 
-	/**
-	 * Checks if the request provided to the constructor is valid.
-	 *
-	 * @return bool True if all prerequisites are met
-	 */
-	protected function setup() {
-		global $wgContLang, $wgCapitalLinks;
+	private function antiSpoofTest( $newUserName ) {
+		if ( class_exists( 'SpoofUser' ) ) {
+			$oNewSpoofUser = new SpoofUser( $newUserName );
+			if ( !$oNewSpoofUser->isLegal() ) {
+				$this->addWarning( wfMessage( 'userrenametool-error-antispoof-conflict', $newUserName ) );
+			}
+		} else {
+			$this->addError( wfMessage( 'userrenametool-error-antispoof-notinstalled' ) );
+		}
+	}
 
-		// Sanitize input data
-		$oldNamePar = trim( str_replace( '_', ' ', $this->mRequestData->oldUsername ) );
-		$oldTitle = Title::makeTitle( NS_USER, $oldNamePar );
+	private function phalanxTest( $userName ) {
+		$warning = RenameUserHelper::testBlock( $userName );
+		if ( !empty( $warning ) ) {
+			$this->addWarning( $warning );
+		}
+	}
 
-		// Force uppercase of newusername, otherwise wikis with wgCapitalLinks=false can create lc usernames
-		$newTitle = Title::makeTitleSafe( NS_USER, $wgContLang->ucfirst( $this->mRequestData->newUsername ) );
+	private function areUserNamesValid( $oldUserName, $newUsername ) {
+		if ( !$this->isUserNameNotEmpty(
+			$oldUserName,
+			'userrenametool-errorinvalid',
+			$this->mRequestData->oldUsername
+		)
+		) {
+			return false;
+		};
+
+		if ( !$this->isUserNameNotEmpty(
+			$newUsername,
+			'userrenametool-errorinvalidnew',
+			$this->mRequestData->newUsername
+		)
+		) {
+			return false;
+		};
 
-		$oun = is_object( $oldTitle ) ? $oldTitle->getText() : '';
-		$nun = is_object( $newTitle ) ? $newTitle->getText() : '';
+		if ( $this->areUserNamesIdentical( $oldUserName, $newUsername ) ) {
+			return false;
+		}
 
-		$this->addInternalLog( "title: old={$oun} new={$nun}" );
+		return true;
+	}
 
-		// AntiSpoof test
+	private function isUserNameNotEmpty( $userName, $errorMessageKey, $errorMessageValue ) {
+		if ( !$userName ) {
```

Comment:
This function is advertises itself as a test for "not empty", however it also has a side effect of adding a message to the error array.  I would consider renaming this to reflect the extra functionality.  However, I'm not sure this abstraction saves much for this one test since every value used has to be passed in.